src/utils/validation/createNonEmptyValidator.js

```diff
@@ -11,8 +11,14 @@ governing permissions and limitations under the License.
 */
 
 import assertValid from "./assertValid";
+import isObject from "../isObject";
+import isEmptyObject from "../isEmptyObject";
 
 export default message => (value, path) => {
-  assertValid(value.length > 0, value, path, message);
+  if (isObject(value)) {
+    assertValid(!isEmptyObject(value), value, path, message);
+  } else {
+    assertValid(value.length > 0, value, path, message);
+  }
```

Comment:
I hadn't really thought deeply about this validator before, but it's interesting that we're using it for strings. I think I'd rather see a `minLength()` and `maxLength()` available for strings where each function accepts a number. For arrays, if we ever need it, we could have `minItems()` and `maxItems()`. These concepts would more closely follow JSON-schema standards, FWIW. Validating that an object has no keys is interesting...`isEmpty()` could work, but we may even be over-validating. I would be fine allowing `xdm` and `data` objects with no keys. Curious to know what others think here.

Reply:
Yes, once we need to validate min or max, it would make sense to add those and use them instead of nonEmpty.